Exporting typesupport libraries does not respect LIBRARY_NAME

## Bug report

- Operating System:
  - Ubuntu 20.04
- Installation type:
  - binaries
- Version or commit hash:
  - ros-foxy-rosidl-typesupport-fastrtps-c: 1.0.4-1focal.20210901.175719
- DDS implementation:
  - Fast-RTPS
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue

Have one package that builds libraries passing in the LIBRARY_NAME variable, have a second that depends on that package and attempts to use the messages.

message package CMakeLists.txt:

```
cmake_minimum_required(VERSION 3.5)
project(make_msgs)
find_package(ament_cmake REQUIRED)
find_package(rosidl_default_generators REQUIRED)
rosidl_generate_interfaces(${PROJECT_NAME}_msgs
        msg/CustomMessage.msg
        LIBRARY_NAME ${PROJECT_NAME}
        )
ament_package()
```

depending package CMakeLists.txt:

```
cmake_minimum_required(VERSION 3.5)
project(depend_msgs)
find_package(ament_cmake REQUIRED)
find_package(make_msgs REQUIRED)
ament_package()
```

Build packages with colcon.

#### Expected behavior

Both package should compile successfully

#### Actual behavior

The depend_msgs package will fail with

```
--- stderr: depend_msgs                         
CMake Error at /testing/ros2/install/make_msgs/share/make_msgs/cmake/rosidl_cmake_export_typesupport_libraries-extras.cmake:26 (message):
  Package 'make_msgs' exports the typesupport library
  'make_msgs_msgs__rosidl_typesupport_fastrtps_c' which couldn't be found
Call Stack (most recent call first):
  /testing/ros2/install/make_msgs/share/make_msgs/cmake/make_msgsConfig.cmake:41 (include)
  CMakeLists.txt:4 (find_package)


---
Failed   <<< depend_msgs [0.34s, exited with code 1]
```

#### Additional information

Checking the `rosidl_cmake_export_typesupport_libraries-extras.cmake` file, the issue is present with the cpp version as well.

This probably does not affect only the fastrtps implementations, but as this is the default rmw for Foxy it is where I ran into the issue.

--- a/rosidl_cmake/cmake/rosidl_export_typesupport_libraries.cmake
+++ b/rosidl_cmake/cmake/rosidl_export_typesupport_libraries.cmake
@@ -45,7 +45,13 @@ macro(rosidl_export_typesupport_libraries library_suffix)
           "not-imported targets")
       endif()
 
-      list(APPEND _ROSIDL_CMAKE_EXPORT_TYPESUPPORT_LIBRARIES "${library_suffix}:${_lib}")
+      get_target_property(_lib_name "${_lib}" OUTPUT_NAME)
+
+      if(NOT _lib_name)
+        set(_lib_name ${_lib})
+      endif()
+
+      list(APPEND _ROSIDL_CMAKE_EXPORT_TYPESUPPORT_LIBRARIES "${library_suffix}:${_lib_name}")
     endforeach()
   endif()
 endmacro()

